utilities/backupable/backupable_db.cc

```diff
@@ -1551,6 +1652,48 @@ Status BackupEngineImpl::CalculateChecksum(const std::string& src, Env* src_env,
   return s;
 }
 
+Status BackupEngineImpl::GetFileDbIdentities(Env* src_env,
+                                             const std::string& file_path,
+                                             std::string* db_id,
+                                             std::string* db_session_id) {
+  // Prepare the full_path of file_path under src_env for SstFileDumper
+  std::string full_path;
+  src_env->GetAbsolutePath(file_path, &full_path);
+
+  SstFileDumper sst_reader(Options(), full_path,
```

Comment:
Does using `Options()` here work generally? I'd think it uses default `env`, `table_factory`, etc. which will be an issue if backing up from a non-default environment, backing up a non-block based table, etc.

Reply:
The tests attempt to catch this bug by using `ChrootEnv` for the source/destination `Env`s so, if we accidentally use the wrong `Env`, there'll be an error (probably missing file). Here, you discovered an interesting way to use the wrong `Env` while satisfying the tests -- get the absolute path using the right `ChrootEnv` and then access it with the default `Env`.

Reply:
This is the same as how sst_dump reads a table file. I think sst_dump first accepts a default option and then gradually fine-tunes the option (to adapt for a specific table type, etc.) I may misunderstand this so please correct me if I'm wrong :P

Reply:
Yes you are correct. We also need to do the same to obtain the correct env, table type, etc. We can use something like `rocksdb::DB::GetOptions()` (I'm not sure yet what to do about column family-specific options as we don't know the column family here...).